.changeset/dirty-socks-sip.md

```diff
@@ -0,0 +1,42 @@
+---
+'@astrojs/markdown-remark': minor
+'astro': minor
+---
+
+Adds a configuration called https://shiki.style/guide/load-lang#custom-language-aliases, that allows a non-supported code language to a known language.
+
+This option requires `langs` to be defined with the correct values. The option will tell shiki which language to load when mapping the alias.
```

Comment:
Digging deeper into this now, it seems like a Shiki bug that requires you to specify the `langs` beforehand for aliases. It might be good to followup upstream later.

Reply:
Well, the docs show that `javascript` is the array

https://shiki.style/guide/load-lang#custom-language-aliases

Reply:
I don't think you have to. You can specify the langs upfront so that you can syntax highlight the code synchronously, but Astro doesn't do that and will always load the langs dynamically, so in practice the `langs` field is only for additional custom languages.

Reply:
Just noting that I came to peek in on this PR and I'm not sure whether this behaviour has been confirmed yet, so I'll wait to review!

Reply:
Is good for docs review now 👍